lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1215,11 +1224,18 @@ def create_default_nic(self):
         self.log("NIC {0} does not exist.".format(network_interface_name))
 
         if self.virtual_network_name:
-            try:
-                self.network_client.virtual_networks.list(self.resource_group, self.virtual_network_name)
-                virtual_network_name = self.virtual_network_name
-            except Exception as exc:
-                self.fail("Error: fetching virtual network {0} - {1}".format(self.virtual_network_name, str(exc)))
+            if self.virtual_network_resource_group:
+                try:
+                    self.network_client.virtual_networks.list(self.virtual_network_resource_group, self.virtual_network_name)
+                    virtual_network_name = self.virtual_network_name
+                except Exception as exc:
```

Comment:
Yes, completely agreed. @ozboms can you use `CloudError` for your error handling here?